src/components/street-align.js

```diff
@@ -0,0 +1,138 @@
+/* global AFRAME */
+
+AFRAME.registerComponent('street-align', {
+  dependencies: ['managed-street'],
+  schema: {
+    width: {
+      default: 'center',
+      type: 'string',
+      oneOf: ['center', 'left', 'right']
+    },
+    length: {
+      default: 'start',
+      type: 'string',
+      oneOf: ['middle', 'start', 'end']
+    }
+  },
+
+  init: function () {
+    // Listen for any segment changes from managed-street
+    this.el.addEventListener('segments-changed', () => this.realignStreet());
+
+    // Initial alignment
+    this.realignStreet();
+
+    // TODO: Still need this for loading from saved scene
+    setTimeout(() => {
+      this.realignStreet();
+    }, 0);
+  },
+
+  onSegmentWidthChanged: function (event) {
+    console.log('segment width changed handler called', event);
+    this.realignStreet();
+  },
+
+  update: function (oldData) {
+    const data = this.data;
+    const diff = AFRAME.utils.diff(oldData, data);
+
+    // Only realign if width or length alignment changed
+    if (diff.width !== undefined || diff.length !== undefined) {
+      this.el.emit('alignment-changed', {
+        changeType: 'alignment',
+        oldData: oldData,
+        newData: data
+      });
+      this.realignStreet();
+    }
+  },
+
+  setupMutationObserver: function () {
```

Comment:
That's dead code, remove.

Reply:
54310b062023f506b3f282a13f7b1e2aac76468e